client/router_create.go

```diff
@@ -736,6 +736,10 @@ func (cli *VanClient) GetRouterSpecFromOpts(options types.SiteConfigSpec, siteId
 // RouterCreate instantiates a VAN (router and controller) deployment
 func (cli *VanClient) RouterCreate(ctx context.Context, options types.SiteConfig) error {
 	// todo return error
+	if options.Spec.IsIngressRoute() && cli.RouteClient == nil {
+		return fmt.Errorf("Routes client had not been initialized, is this an OCP cluster?")
```

Comment:
orr added "Also" to "siteConfigCreate"? or to the cli "Run" function?
there is a logic there in the cli "Run" function that I do not fully understand... 

```siteConfig, err := cli.SiteConfigInspect(context.Background(), nil)
                        if err != nil {
                                return err
                        }

                        if siteConfig == nil {
                                siteConfig, err = cli.SiteConfigCreate(context.Background(), routerCreateOpts)
                                if err != nil {
                                        return err
                                }
                        }
                        // IF siteConfig != nil ? routerCreateOpts is completely ignored?

```

Reply:
Sorry, missed this comment earlier. Unless the skupper-site config map already exists, siteConfig will be nil and the config map will then be created using routerCreateOpts. If it does exist, the settings in the config map should be used. This is for the case that RouterCreate is called by the site-controller instead of the cli.